modules/vstudio/vs2010_vcxproj_user.lua

```diff
@@ -54,8 +54,12 @@
 
 	function m.userProject()
 		local action = p.action.current()
-		p.push('<Project ToolsVersion="%s" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">',
-			action.vstudio.toolsVersion)
+		if _ACTION >= "vs2019" then
+			p.push('<Project ToolsVersion="Current" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">')
```

Comment:
Why couldn't the `toolsVersion` in the VS2019 action be set to `"Current"` instead? The point of the `toolsVersion` is to avoid multiple code paths, so it would be good to maintain this if we can.

Reply:
Interesting, perhaps allowing `action.vstudio.toolsVersion` to be a function? I'm fairly certain we have this in a few other places, checking the type is a function, otherwise assuming it's the expected type. The functions tend to have a single parameter `wks`, `prj` or `cfg` whichever is appropriate. This would allow you to modify the usages of `action.vstudio.toolsVersion` but only convert VS2019 to a function.

Reply:
Yea okay, but would wks, prj, or cfg know that its making a vcxproj.user file? My inclination would be it isn't, since the function would basically have to do something like so:
```
if projectType == "user" then
    return "Current"
else
    return "15.0"
end
```
It would be "simpler" to just pass in whether it is the `vcxproj.user` file as a single parameter.

Reply:
An alternative could be if we changed toolsVersion into a table... So instead we could have
```
toolsVersion = {
    project = "15.0",
    user = "Current"
}
```
The downside would be we would need to change all the instances where toolsVersion was used. More up-front work to get it working, but it would solve the problem relatively elegantly.

Or we just change the `vstudio` table to have a `userToolsVersion` variable, although the result would be the same.

Reply:
I came up with a better alternative, I just added a table value to 2019 only, userToolsVersion, which is just optionally checked by this:
```
p.push('<Project ToolsVersion="%s" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">',
				action.vstudio.userToolsVersion or action.vstudio.toolsVersion)
```
Hope this resolves the issue for you

Reply:
@samsinsane ping

Reply:
You could have used `project.isdotnet(prj)` in the function to swap between the values. However, I think splitting them into the two keys in the action is probably for the best since there is actually three values; nothing in `vcxproj` (omitting the `ToolsVersion` attribute), `Current` in the .NET projects, and `15.0` in `vcxproj.user` (Shouldn't this be `16.0`? VS2017 is 15.0). Your way is the least complicated, and ultimately is far more flexible if Microsoft decide to read `ToolsVersion` to the `vcxproj` in VS2021.